### PR TITLE
Union #818 #821 #822

### PR DESCRIPTION
**Pull request by jepio**
## Description

This is a merge of:
- [x] #818 
- [ ] #821 
- [x] #822 

This is intended to demonstrate that moduletests pass when all three PRs are merged together. Not intended for merging.

## Checklist

- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-osconfig/blob/main/CONTRIBUTING.md).
- [ ] I added unit-tests to validate my changes. All unit tests are passing.
- [ ] I have merged the latest `main` branch prior to this PR submission.
- [ ] I submitted this PR against the `main` branch.

### CODE REVIEW

File: src/modules/test/CMakeLists.txt [hunk 1]
```diff
@@ -23,4 +23,19 @@ target_include_directories(moduletest PRIVATE ${MODULES_INC_DIR} ${CMAKE_CURRENT
 add_custom_command(TARGET moduletest POST_BUILD
     COMMAND ${CMAKE_COMMAND} -E create_symlink $<TARGET_FILE:moduletest> ${CMAKE_BINARY_DIR}/moduletest
     DEPENDS $<TARGET_FILE:moduletest>
-)
\ No newline at end of file
+)
+
```
Comment (AhmedBM) [inline src/modules/test/CMakeLists.txt:27]: nit: extra newlines

File: src/modules/test/CMakeLists.txt [hunk 2]
```diff
@@ -23,4 +23,19 @@ target_include_directories(moduletest PRIVATE ${MODULES_INC_DIR} ${CMAKE_CURRENT
 add_custom_command(TARGET moduletest POST_BUILD
     COMMAND ${CMAKE_COMMAND} -E create_symlink $<TARGET_FILE:moduletest> ${CMAKE_BINARY_DIR}/moduletest
     DEPENDS $<TARGET_FILE:moduletest>
-)
\ No newline at end of file
+)
+
+
+set(SRC  ${moduletest_SOURCE_DIR}/recipes)
+set(MOF1 ${OsConfigResourceSsh_SOURCE_DIR}/LinuxSshServerSecurityBaseline.mof)
```
Comment (AhmedBM) [inline src/modules/test/CMakeLists.txt:30]: a CMake `list` might be better here as the list grows...
eg.
```cmake
set(MOFS
  ${OsConfigResourceSsh_SOURCE_DIR}/LinuxSshServerSecurityBaseline.mof
  ${OsConfigResourceAsb_SOURCE_DIR}/AzureLinuxBaseline.mof
)
list(JOIN MOF_LIST " " flat_string)
add_custom_command(
    OUTPUT ${SRC}/SecurityBaselineTests.json
    DEPENDS ${SRC}/create-asb-json.sh ${SRC}/mof-to-json.awk ${SRC}/SecurityBaselineTests.json-header ${SRC}/SecurityBaselineTests.json-mid ${SRC}/SecurityBaselineTests.json-footer ${MOF1} ${MOF2}
    COMMAND ./create-asb-json.sh ${MOF_LIST} >${SRC}/SecurityBaselineTests.json
    WORKING_DIRECTORY ${SRC}
)
```

**Review by AhmedBM — COMMENTED** [review]
Just a few minor comments but @MariusNi would have to sign off on this but FYI hes out for the week

File: src/common/commonutils/DeviceInfoUtils.c [hunk 3]
```diff
@@ -947,4 +947,22 @@ bool IsCommodore(void* log)
     FREE_MEMORY(textResult);
 
     return status;
-}
\ No newline at end of file
+}
+
+enum SelinuxState {
```
Comment (MariusNi) [inline src/common/commonutils/DeviceInfoUtils.c:952]: >SelinuxState [](http://example.com/codeflow?start=5&length=12)

nit, fix casing: 'SeLinuxState'

File: src/common/commonutils/DeviceInfoUtils.c [hunk 4]
```diff
@@ -947,4 +947,22 @@ bool IsCommodore(void* log)
...
     return status;
-}
\ No newline at end of file
+}
+
+enum SelinuxState {
+  SelinuxUnknown = 0,
+  SelinuxFound,
+  SelinuxNotFound,
+};
```
Comment (MariusNi) [inline src/common/commonutils/DeviceInfoUtils.c:956]: Do all explicit initializations and complete with a typedef like this, for example:

```
enum PasswordEncryption
{
    unknown = 0,
    md5 = 1,
    blowfish = 2,
    eksBlowfish = 3,
    unknownBlowfish = 4,
    sha256 = 5,
    sha512 = 6
};
typedef enum PasswordEncryption PasswordEncryption;
```

File: src/common/commonutils/FileUtils.c [hunk 5]
```diff
@@ -893,6 +916,11 @@ int RenameFile(const char* original, const char* target, void* log)
         status = (0 == errno) ? ENOENT : errno;
     }
 
+    if (DetectSelinux(log))
+    {
+        RestoreSelinuxContext(target, log);
+    }
```
Comment (MariusNi) [inline src/common/commonutils/FileUtils.c:922]: Can you add a one line comment that describes why this is necessary?

File: src/common/commonutils/PackageUtils.c [hunk 6]
```diff
@@ -187,6 +209,7 @@ int InstallOrUpdatePackage(const char* packageName, void* log)
     
     if (g_aptGetIsPresent)
     {
+        AptGetUpdateOnce(log);
```
Comment (MariusNi) [inline src/common/commonutils/PackageUtils.c:212]: >AptGetUpdateOnce(log); [](http://example.com/codeflow?start=8&length=22)

How long can this take? There is a concern of ending in Azure Policy timeout on slow machines due to lengthy package operations during audit

File: src/modules/test/recipes/mof-to-json.awk [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+        map[key] = 1;
+    }
+}
+
+# Print the final output
+END {
+    # Remove the last comma for valid JSON format
+    sub(/,$/, "", output);
+    printf output;
+}
```
Comment (MariusNi) [inline src/modules/test/recipes/mof-to-json.awk:104]: nit: line 105 is blank, remove

File: src/modules/test/recipes/create-asb-json.sh [hunk 8]
```diff
@@ -0,0 +1,11 @@
...
+
+template=SecurityBaselineTests.json
+
+(
+  cat "${template}-header";
+  ./mof-to-json.awk -v remediate=1 "${@}"
+  cat "${template}-mid";
+  ./mof-to-json.awk -v remediate=0 "${@}"
+  cat "${template}-footer";
+)
```
Comment (MariusNi) [inline src/modules/test/recipes/create-asb-json.sh:11]: Why split recipe in footer, header and mid parts? In general test recipes can contain any number of reload module sequences. If we'll add another to ASB's recipe, we'll then add a new mid2 etc? Please consider putting this back into a single test recipe JSON .

File: src/tests/e2e-test-recipes/DeviceInfoTests.json [hunk 9]
```diff
@@ -1,102 +0,0 @@
-[
-  {
-    "Action": "LoadModule",
-    "Module": "deviceinfo.so"
-  },
-  {
```
Comment (MariusNi) [inline src/tests/e2e-test-recipes/DeviceInfoTests.json:6]: Please do not change these test recipes. We have lots more in temporarily disabled modules and not all align with this model. A test recipe can contain anything, there is no requirement to start with a certain header, etc. We do not want to put such restrictions.

Please consider only changing in the ASB recipe and keeping that as one single JSON file as it is.

**Pull request closed without merging**